Library/Homebrew/cmd/doctor.rb

```diff
@@ -435,6 +443,7 @@ def check_access_usr_local
 
     You should probably change the ownership and permissions of /usr/local
     back to your user account.
+      sudo chown $(whoami):admin /usr/local
```

Comment:
You should see warning message when running `brew update`(instead of manually `git pull`). 

As for `brew repair-permission`(though it hasn't been implemented), it should repair permissions for these files:
* `HOMEBREW_REPOSITORY` directory
* Every files in `git ls-tree`
* Every files and directories in `HOMEBREW_LIBRARY`
* `HOMEBREW_PREFIX` directory
* `HOMEBREW_PREFIX/opt` directory. 
* All directories and their subdirectories of `Keg::TOP_LEVEL_DIRECTORIES`
* `Language::Python.homebrew_site_package`
* Every files and directories in `HOMEBREW_CELLAR`
* `HOMEBREW_LOGS`
* `HOMEBREW_CACHE`
* `HOMEBREW_TEMP`